testdata/p4tc_samples_outputs/ipip_control_blocks.c

```diff
@@ -182,7 +182,6 @@ if (/* hdr->outer.isValid() */
                 return TC_ACT_SHOT;
             }
             
-            hdr->ethernet.dstAddr = htonll(hdr->ethernet.dstAddr << 16);
```

Comment:
I found an issue which is not shown in any example you have here, however if you remove the `@tc_type("macaddr")` annotation from the ipip ethernet header (https://github.com/komaljai/p4c/blob/networkOrder_arithmetic/testdata/p4tc_samples/ipip.p4#L11) you'll see it

When it converts the mac address to host endian (note that in this case there will be no annotation in the header) https://github.com/komaljai/p4c/blob/networkOrder_arithmetic/testdata/p4tc_samples_outputs/ipip_control_blocks.c#L104, it will do this:

```
hdr->ethernet.dstAddr = ntohll(value->u.Main_set_nh.dmac);
```

However the deparser will try to convert back shifting, as in the line above:

```
hdr->ethernet.dstAddr = htonll(hdr->ethernet.dstAddr << 16);
```

So the mac address will be written in the deparser skipping a couple bytes